pycode/memilio-surrogatemodel/memilio/surrogatemodel/GNN/data_generation_nodamp.py

```diff
@@ -0,0 +1,328 @@
+            date(start_year, start_month, start_day) - date(start_year, 1, 1)).days
+
+        # Load baseline and minimum contact matrix and assign them to the model
+        baseline = getBaselineMatrix()
+        #minimum = getMinimumMatrix()
+
+        model.parameters.ContactPatterns.cont_freq_mat[0].baseline = baseline
+        model.parameters.ContactPatterns.cont_freq_mat[0].minimum = np.ones(
+        (num_groups, num_groups)) * 0
+
+        # Apply mathematical constraints to parameters
+        model.apply_constraints()
+        models.append(model)
+
+    graph = ModelGraph()
+    for i in range(num_regions):
+        graph.add_node(int(countykey_list[i]), models[i])
+
+    # get mobility data directory
+    arg_dict = gd.cli("commuter_official")
+
+    directory = arg_dict['out_folder'].split('/pydata')[0]
+    directory = os.path.join(directory, 'mobility/')  
+
+    # Merge Eisenach and Wartbugkreis in Input Data 
+    tmd.updateMobility2022(directory, mobility_file='twitter_scaled_1252')
+    tmd.updateMobility2022(directory, mobility_file='commuter_migration_scaled')
+
+    num_locations = 4
+
+    set_edges(os.path.abspath(os.path.join(directory, os.pardir)), 
+                            graph, num_locations)
+
+    study = ParameterStudy(graph, 0, days, dt=dt, num_runs=1)
+    study.run()
+
+    graph_run = study.run()[0]
+    results = interpolate_simulation_result(graph_run)
+
+    for result_indx in range(len(results)):
+        results[result_indx] = remove_confirmed_compartments(
+            np.transpose(results[result_indx].as_ndarray()[1:, :]), num_groups)
+
+    # Omit first column, as the time points are not of interest here.
+    dataset_entry = copy.deepcopy(results)
+
+    return dataset_entry
+
+def remove_confirmed_compartments(dataset_entries, num_groups):
+    """! The compartments which contain confirmed cases are not needed and are 
+        therefore omitted by summarizing the confirmed compartment with the 
+        original compartment. 
+    @param dataset_entries Array that contains the compartmental data with 
+            confirmed compartments. 
+    @param num_groups Number of age groups.
+    @return Array that contains the compartmental data without confirmed compartments. 
+   """
+    
+    new_dataset_entries = []
+    for i in dataset_entries : 
+      dataset_entries_reshaped  = i.reshape(
+          [num_groups, int(np.asarray(dataset_entries).shape[1]/num_groups)]
+          )
+      sum_inf_no_symp = np.sum(dataset_entries_reshaped [:, [2, 3]], axis=1)
+      sum_inf_symp = np.sum(dataset_entries_reshaped [:, [4, 5]], axis=1)
+      dataset_entries_reshaped[:, 2] = sum_inf_no_symp
+      dataset_entries_reshaped[:, 4] = sum_inf_symp
+      new_dataset_entries.append(
+          np.delete(dataset_entries_reshaped , [3, 5], axis=1).flatten()
+          )
+    return new_dataset_entries
+
+
+def get_population(path="data/pydata/Germany/county_population.json"):
```

Comment:
we have a predefined & unti-tested function in the epidata package doing this called `get_population_data`. We should definitely use this then.